polyfactory/utils/helpers.py

```diff
@@ -13,6 +13,14 @@
     from random import Random
     from typing import Sequence
 
+    from typing_extensions import get_args, get_origin
+else:
+    try:
+        # get_args function not returning the origin type as expected for pydantic v1 constrained values, ex. ConstrainedListValue.
+        from pydantic.utils import get_args, get_origin
+    except ImportError:
+        from typing_extensions import get_args, get_origin
+
```

Comment:
Could we remove this for now? I think this should be handled seperately in a different issue/PR. I would like to explore if there's a better way to do this (my gut feeling is we'll probably have to end up doing this as a solution anyway, but I'd still like to check).

Reply:
Sure, I kept a test for that `test_factory_use_construct_nested_constraint_list_v1` as skipped, because we are going to fix this. If you want, I can also delete it.